ttnn/cpp/ttnn/operations/core/compute_kernel/compute_kernel_config.cpp

```diff
@@ -35,18 +35,27 @@ DeviceComputeKernelConfig init_device_compute_kernel_config(
                         .math_approx_mode = math_approx_mode,
                         .dst_full_sync_en = dst_full_sync_en};
                 } else if constexpr (std::is_same_v<T, WormholeComputeKernelConfig>) {
-                    TT_ASSERT(ttnn::device::is_wormhole_or_blackhole(arch), "kernel config is not for wormhole_b0 or blackhole");
-                    MathFidelity math_fidelity = compute_kernel_config.math_fidelity;
-                    bool math_approx_mode = compute_kernel_config.math_approx_mode;
-                    bool fp32_dest_acc_en = compute_kernel_config.fp32_dest_acc_en;
-                    bool packer_l1_acc = compute_kernel_config.packer_l1_acc;
-                    bool dst_full_sync_en = compute_kernel_config.dst_full_sync_en;
-                    defaultConfig = WormholeComputeKernelConfig{
-                        .math_fidelity = math_fidelity,
-                        .math_approx_mode = math_approx_mode,
-                        .fp32_dest_acc_en = fp32_dest_acc_en,
-                        .packer_l1_acc = packer_l1_acc,
-                        .dst_full_sync_en = dst_full_sync_en};
+                    if(ttnn::device::is_wormhole_or_blackhole(arch)) {
+                        MathFidelity math_fidelity = compute_kernel_config.math_fidelity;
+                        bool math_approx_mode = compute_kernel_config.math_approx_mode;
+                        bool fp32_dest_acc_en = compute_kernel_config.fp32_dest_acc_en;
+                        bool packer_l1_acc = compute_kernel_config.packer_l1_acc;
+                        bool dst_full_sync_en = compute_kernel_config.dst_full_sync_en;
+                        defaultConfig = WormholeComputeKernelConfig{
+                            .math_fidelity = math_fidelity,
+                            .math_approx_mode = math_approx_mode,
+                            .fp32_dest_acc_en = fp32_dest_acc_en,
+                            .packer_l1_acc = packer_l1_acc,
+                            .dst_full_sync_en = dst_full_sync_en};
+                    } else if(arch == tt::ARCH::GRAYSKULL) {
```

Comment:
pls remove this block grayskull is handled at line 28
```suggestion
                    } else if(arch == tt::ARCH::GRAYSKULL) {
```